src/livecodes/html/templates.html

```diff
@@ -28,3 +32,24 @@
     </div>
   </div>
 </div>
+
+<!-- Inline: emit a "templates:filter" event with the current query so other scripts can perform filtering -->
+<script>
+  (function () {
+    var input = document.getElementById('templates-search-input');
+    if (!input) return;
+
+    var emit = function (value) {
+      var ev = new CustomEvent('templates:filter', { detail: { query: value } });
+      document.getElementById('templates-container').dispatchEvent(ev);
+    };
+
+    var timeout = null;
+    input.addEventListener('input', function (e) {
+      var val = e.target.value || '';
+      // debounce to avoid excessive events while typing
+      clearTimeout(timeout);
+      timeout = setTimeout(function () { emit(val.trim()); }, 150);
```

Comment:
There is a [debounce function here](https://github.com/live-codes/livecodes/blob/7344c7883533e904543fbcac4b983f5a3d5880b8/src/livecodes/utils/utils.ts#L4) that you can re-use.